Crash in crypto subsystem

```[INFO] 2018/01/31 09:21:10 Starting UDP hole punching to 158.181.16.71:34450
panic: crypto/cipher: input not full blocks

goroutine 82 [running]:
crypto/cipher.(*cbcDecrypter).CryptBlocks(0xc4202e87d0, 0xc4201adc30, 0x14, 0x14, 0xc4201adc30, 0x14, 0x14)
	/usr/lib/go/src/crypto/cipher/cbc.go:114 +0x498
github.com/subutai-io/p2p/lib.Crypto.decrypt(0xc4203761e0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0xbe946ee0e3e32f94, 0x34a71f1cecc, 0xc2c220, ...)
	/home/crioto/Projects/golang/src/github.com/subutai-io/p2p/lib/crypto.go:103 +0x11e
github.com/subutai-io/p2p/lib.(*PeerToPeer).HandleP2PMessage(0xc420210dc0, 0x36, 0xc420304d50, 0x0, 0x0, 0xc42025a630, 0x1000, 0x1000)
	/home/crioto/Projects/golang/src/github.com/subutai-io/p2p/lib/packet_handler.go:32 +0x349
github.com/subutai-io/p2p/lib.(*PeerToPeer).HandleP2PMessage-fm(0x36, 0xc420304d50, 0x0, 0x0, 0xc42025a630, 0x1000, 0x1000)
	/home/crioto/Projects/golang/src/github.com/subutai-io/p2p/lib/p2p.go:274 +0x73
github.com/subutai-io/p2p/lib.(*Network).Listen(0xc42025a600, 0xc42020f200)
	/home/crioto/Projects/golang/src/github.com/subutai-io/p2p/lib/net.go:395 +0xdb
created by github.com/subutai-io/p2p/lib.New
	/home/crioto/Projects/golang/src/github.com/subutai-io/p2p/lib/p2p.go:274 +0x8bc
```

--- a/lib/crypto.go
+++ b/lib/crypto.go
@@ -5,6 +5,7 @@ import (
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/rand"
+	"fmt"
 	"io/ioutil"
 	"strconv"
 	"time"
@@ -98,7 +99,9 @@ func (c Crypto) decrypt(key []byte, data []byte) ([]byte, error) {
 		return nil, err
 	}
 	encData := data[aes.BlockSize:]
-
+	if len(data)%aes.BlockSize != 0 {
+		return nil, fmt.Errorf("Input not full blocks: %s", string(data))
+	}
 	mode := cipher.NewCBCDecrypter(block, data[:aes.BlockSize])
 	mode.CryptBlocks(encData, encData)
 

--- a/lib/packet_handler.go
+++ b/lib/packet_handler.go
@@ -31,7 +31,8 @@ func (p *PeerToPeer) HandleP2PMessage(count int, srcAddr *net.UDPAddr, err error
 		var decErr error
 		msg.Data, decErr = p.Crypter.decrypt(p.Crypter.ActiveKey.Key, msg.Data)
 		if decErr != nil {
-			Log(Error, "Failed to decrypt message")
+			Log(Error, "Failed to decrypt message: %s", decErr)
+			return
 		}
 		msg.Data = msg.Data[:msg.Header.Length]
 
@@ -57,7 +58,7 @@ func (p *PeerToPeer) HandlePingMessage(msg *P2PMessage, srcAddr *net.UDPAddr) {
 	// p.proxyLock.Lock()
 	// defer p.proxyLock.Unlock()
 	if err != nil {
-		if p.ProxyManager.touch(string(msg.Data)) {
+		if p.ProxyManager.touch(srcAddr.String()) {
 			p.UDPSocket.SendMessage(msg, srcAddr)
 		}
 		// for i, proxy := range p.Proxies {

--- a/lib/proxy.go
+++ b/lib/proxy.go
@@ -79,14 +79,17 @@ func (p *ProxyManager) check() {
 			if err != nil {
 				Log(Debug, "Failed to close proxy: %s", err)
 			}
+			Log(Debug, "Failed to connect to proxy %s", id)
 		}
 		if proxy.Status == proxyActive && time.Since(proxy.LastUpdate) > time.Duration(30*time.Second) {
 			err := proxy.Close()
 			if err != nil {
 				Log(Debug, "Failed to close proxy: %s", err)
 			}
+			Log(Debug, "Proxy %s has been disconnected by timeout", id)
 		}
 		if proxy.Status == proxyDisconnected {
+			Log(Debug, "Removing proxy %s", id)
 			p.operate(OperateDelete, id, nil)
 		}
 	}

